spark/src/main/scala/org/apache/spark/whylogs/DatasetProfileMerger.scala

```diff
@@ -0,0 +1,34 @@
+package org.apache.spark.whylogs
+
+import com.whylogs.core.DatasetProfile
+import com.whylogs.spark.WhyLogs.PROFILE_FIELD
+import org.apache.spark.sql.catalyst.encoders.ExpressionEncoder
+import org.apache.spark.sql.expressions.Aggregator
+import org.apache.spark.sql.{Encoder, Encoders, Row}
+
+import java.io.ByteArrayInputStream
+import java.time.Instant
+import java.util.UUID
+
+class DatasetProfileMerger(datasetName: String,
+                           sessionTimeInMillis: Long,
+                           sessionId: String = UUID.randomUUID().toString) extends Aggregator[Row, DatasetProfile, Array[Byte]] {
+  override def zero: DatasetProfile =  new DatasetProfile(sessionId, Instant.ofEpochMilli(sessionTimeInMillis))
+    .withTag("Name", datasetName)
+
+  override def reduce(b: DatasetProfile, row: Row): DatasetProfile = {
+    val is = new ByteArrayInputStream(row.getAs[Array[Byte]](PROFILE_FIELD));
+    val rhs = DatasetProfile.parse(is)
+    b.merge(rhs)
+  }
+
```

Comment:
Not sure if its necessary here, but most of the UDAF's where I've done profile merging end up with a pattern something like

    if (left == null || left.length == 0) {
      return right;
    }
    if (right == null || right.length == 0) {
      return left;
    }
// Then merge